common/src/main/java/com/wynntils/models/objectives/DailyObjectiveScoreboardPart.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.models.objectives;
+
+import com.wynntils.core.components.Models;
+import com.wynntils.handlers.scoreboard.ScoreboardSegment;
+import com.wynntils.handlers.scoreboard.type.SegmentMatcher;
+import java.util.List;
+
+public class DailyObjectiveScoreboardPart extends AbstractObjectivesScoreboardPart {
+    private static final SegmentMatcher OBJECTIVES_MATCHER =
+            SegmentMatcher.fromPattern("([★⭑] )?(Daily )?Objectives?:");
+
+    @Override
+    public SegmentMatcher getSegmentMatcher() {
+        return OBJECTIVES_MATCHER;
+    }
+
+    @Override
+    public void onSegmentChange(ScoreboardSegment newValue) {
+        List<WynnObjective> objectives = parseObjectives(newValue).stream()
+                .filter(wynnObjective -> wynnObjective.getScore().current()
+                        < wynnObjective.getScore().max())
+                .toList();
+
+        for (WynnObjective objective : objectives) {
+            if (!objective.isGuildObjective()) {
+                Models.Objectives.updatePersonalObjective(objective);
+            }
+        }
+
+        // filter out deleted objectives
+        Models.Objectives.purgePersonalObjectives(objectives);
+    }
+
+    @Override
+    public String toString() {
```

Comment:
Do we need a toString for this and the other scoreboard part classes?

Reply:
Not strictly. It is more like a debugging measurement. Just like how many item related classes do.

Reply:
Since it has no fields, the default toString will still print the class name, I think that's fine, and it makes the code a bit more compact.

Reply:
It will print the fully qualified name though.. So `com.wynntils.models.objectives.DailyObjectiveScoreboardPart`.

Reply:
Ok, fine, let's keep it.